### PR TITLE
[server] Change response codes returned from `api/feature-flags/slow-database` endpoint

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description

Make the `api/feature-flags/slow-database` return code `200` and set the `X-Gitpod-Slow-Database` header consistently.

We don't need to distinguish between response codes for the reverse proxy 'pre-flight' added in #14897  and in fact having the endpoint return non-success codes is actively unhelpful when using the Caddy [forward_auth](https://caddyserver.com/docs/caddyfile/directives/forward_auth) directive.

See https://github.com/gitpod-io/gitpod/pull/14897#discussion_r1030654402 for more context. 

## Related Issue(s)

Part of #9198 

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/server/src/feature-flag/featureflag-controller.ts [hunk 1]
```diff
@@ -24,7 +24,8 @@ export class FeatureFlagController {
     protected addSlowDatabaseFeatureFlagHandler(router: express.Router) {
         router.get("/slow-database", async (req, res) => {
             if (!User.is(req.user)) {
-                res.sendStatus(401);
+                res.setHeader("X-Gitpod-Slow-Database", "false");
```
Comment (easyCZ) [inline components/server/src/feature-flag/featureflag-controller.ts:27]: I think you could write it as this:

```

func getSlowDBValue(user) bool {
	if (!User.is(req.user)) {
       return false
    }

	return await getExperimentsClientForBackend().getValueAsync("slow_database", false, {
	        user: req.user,
	    });
}

value = getSlowDBValue(req.user)
res.setHeader("X-Gitpod-Slow-Database", "false");
res.sendStatus(200);
```

Comment (easyCZ) [reply components/server/src/feature-flag/featureflag-controller.ts:27]: Alternatively, if the try catch is needed, then use finally to perform the res & end part of the handling.

**Review by easyCZ — APPROVED** [review]

**Review by geropl — APPROVED** [review]
LGTM :heavy_check_mark:

File: components/server/src/feature-flag/featureflag-controller.ts (hunk 1 above)
Comment (andrew-farries) [reply components/server/src/feature-flag/featureflag-controller.ts]: changed to use `finally`.

File: components/server/src/feature-flag/featureflag-controller.ts [hunk 2]
```diff
@@ -23,21 +23,21 @@ export class FeatureFlagController {
...
     protected addSlowDatabaseFeatureFlagHandler(router: express.Router) {
         router.get("/slow-database", async (req, res) => {
-            if (!User.is(req.user)) {
-                res.sendStatus(401);
-                return;
-            }
-
             try {
+                if (!User.is(req.user)) {
+                    res.setHeader("X-Gitpod-Slow-Database", "false");
```
Comment (AlexTugarev) [inline components/server/src/feature-flag/featureflag-controller.ts:28]: Nice catch!

**Pull request merged**